CHANGES.md

```diff
@@ -0,0 +1,2049 @@
+# Changes
+
+[//]: # **DEFINE LATEST CHANGES UNDER BELOW 'Unreleased' SECTION - THEY WILL BE INTEGRATED IN NEXT RELEASE**
+
+## [Unreleased](https://github.com/bird-house/birdhouse-deploy/tree/master) (latest)
+
+[//]: # list changes here, using '-' for each new entry (remove this when items are added)
```

Comment:
So should you be summarizing this PR in here so next time we run the bump2release command the changelog is already up-to-date?

Going forward, when opening a PR we should also include changelog update in the same PR if I understand correctly?

Another "procedure" question.  Before, I used to tag after the PR is merged, since there is no new commit involved.  Now with bump2version, it will modify a few files, so should the bump happen when the PR is approved and right before the PR is merged?  Otherwise, it will cost another PR to merge the modified files!

Reply:
Yes, I actually need to add my changelog update here. 
It should be added in the same PR. I will add it too in the following commit.

For the tag, this is actually up to us to define how we want to proceed. Some possible alternatives: 

1) We bump in the PR itself as final commit before merge. The tag would be applied one commit before the merge-commit. 
Personally, I don't like this too much, because if we had another commit to the PR after bumping by mistake (or to patch something), it will be technically left out of the tagged version.

2) We modify `.bumpversion.cfg` option `tag = true` to `tag = false`. We then run `bump2version` as normal toward the end of the PR, but the tag is not created. After merge, we push the corresponding tag manually onto the merge-commit.
The downside of this could be that 2 PRs being worked on at the same time and about to be merged will cause some version conflict. The 2nd PR would have to push another commit to patch the erroneous version to one above.

3) We proceed like I do in Magpie, the PR is merged as is without bump. Everything remains  in "unreleased" until we decide to bump, which can be following every single PR, or after an aggregation of a few features as desired. When we decide to "release" a tag with one or more features, we run bump2version on the merged master and push directly the update. Since only versioning items in README and CHANGES files are modified, I don't think it is too critical to do this, but I can understand if there is opposition to do it this way.

Let me know which (or other?) method you would prefer since you will be doing it most of the time.

Reply:
Option 3 (all release activities after the merge) is most similar to what I do now.  But given this repo directly impact deployment, we have set a restriction to not allow direct push to `master`, requiring PR at all time and at least one approuver to merge the PR.  This option means another PR !

Option 1, like you said, might put the tag not at the end of the PR, diffing between tags later will be error prone.

I guess we are left with Option 2.  By the time we have too frequent PR merge race we can re-consider Option 3.

Reply:
Sounds good. I'll push the update for option 2.